src/Symfony/Component/DependencyInjection/Tests/Compiler/ValidateEnvPlaceholdersPassTest.php

```diff
@@ -0,0 +1,265 @@
+     */
+    public function testNulledEnvInConfig()
+    {
+        $container = new ContainerBuilder();
+        $container->setParameter('env(NULLED)', null);
+        $container->registerExtension(new EnvExtension());
+        $container->prependExtensionConfig('env_extension', array(
+            'int_node' => '%env(NULLED)%',
+        ));
+
+        $this->doProcess($container);
+    }
+
+    public function testValidateEnvOnMerge()
+    {
+        $container = new ContainerBuilder();
+        $container->registerExtension($ext = new EnvExtension());
+        $container->prependExtensionConfig('env_extension', array(
+            'int_node' => '%env(int:const:FOO)%',
+            'bool_node' => true,
+        ));
+        $container->prependExtensionConfig('env_extension', array(
+            'int_node' => '%env(int:BAR)%',
+            'bool_node' => '%env(bool:int:BAZ)%',
+            'scalar_node' => '%env(BAZ)%',
+        ));
+
+        $this->doProcess($container);
+
+        $expected = array(
+            'int_node' => '%env(int:const:FOO)%',
+            'bool_node' => true,
+            'scalar_node' => '%env(BAZ)%',
+        );
+
+        $this->assertSame($expected, $container->resolveEnvPlaceholders($ext->getConfig()));
+    }
+
+    public function testConcatenatedEnvInConfig()
+    {
+        $container = new ContainerBuilder();
+        $container->registerExtension($ext = new EnvExtension());
+        $container->prependExtensionConfig('env_extension', array(
+            'scalar_node' => $expected = 'foo %env(BAR)% baz',
+        ));
+
+        $this->doProcess($container);
+
+        $this->assertSame(array('scalar_node' => $expected), $container->resolveEnvPlaceholders($ext->getConfig()));
+    }
+
+    /**
+     * @expectedException \Symfony\Component\Config\Definition\Exception\InvalidConfigurationException
+     * @expectedExceptionMessage A dynamic value is not compatible with a "Symfony\Component\Config\Definition\EnumNode" node type at path "env_extension.enum_node".
+     */
+    public function testEnvIsIncompatibleWithEnumNode()
+    {
+        $container = new ContainerBuilder();
+        $container->registerExtension(new EnvExtension());
+        $container->prependExtensionConfig('env_extension', array(
+            'enum_node' => '%env(FOO)%',
+        ));
+
+        $this->doProcess($container);
+    }
+
+    /**
+     * @expectedException \Symfony\Component\Config\Definition\Exception\InvalidConfigurationException
+     * @expectedExceptionMessage A dynamic value is not compatible with a "Symfony\Component\Config\Definition\ArrayNode" node type at path "env_extension.simple_array_node".
+     */
+    public function testEnvIsIncompatibleWithArrayNode()
+    {
+        $container = new ContainerBuilder();
+        $container->registerExtension(new EnvExtension());
+        $container->prependExtensionConfig('env_extension', array(
+            'simple_array_node' => '%env(json:FOO)%',
```

Comment:
`json:` can return a simple_array, e.g. in the case the env is something like `[1, 2]` - of course `json` doesnt guarrentee that its like this, it could also be `{"a": 1, "b": 2}`, so its a bit of a question, is it valid or not - I think its more useful if it passes validation.

Reply:
yeah i think we need to reassure / revise integration a bit. Or put different im curious if the change to json prefix affects the possibilities here.

Reply:
the json change only added the possibility to accept `null` in addition to `array` so shouldn't change anything here - this was on purpose so that its still possible to do some validation based on it always being an array.

I wonder if it would make sense to add a `simple_array:` decoder to further help with validation